library/std/src/os/fd/owned.rs

```diff
@@ -170,17 +170,24 @@ impl Drop for OwnedFd {
     #[inline]
     fn drop(&mut self) {
         unsafe {
-            // Note that errors are ignored when closing a file descriptor. The
-            // reason for this is that if an error occurs we don't actually know if
-            // the file descriptor was closed or not, and if we retried (for
-            // something like EINTR), we might close another valid file descriptor
-            // opened after we closed ours.
             #[cfg(not(target_os = "hermit"))]
             {
                 #[cfg(unix)]
                 crate::sys::fs::debug_assert_fd_is_open(self.fd);
 
-                let _ = libc::close(self.fd);
+                match cvt(libc::close(self.fd)) {
+                    Ok(_) => {},
+                    Err(err) if err.raw_os_error() == Some(libc::EBADF) {
+                        panic!("`OwnedFd` held an invalid file descriptor");
```

Comment:
`debug_assert_fd_is_open` (added in https://github.com/rust-lang/rust/pull/124210) checks if the file descriptor actually existed. The reason that PR doesn't just check the result of `close` is because a FUSE filesystem can return `EBADF` from `close` even if the fd actually exists.

Reply:
Hmmmmm. I see.

It's sad that there's no way to return an error here, would be nice if the user could somehow be alerted to this problem even in release mode.

Reply:
the right way to do that is to add an explicit `fn close(self) -> Result` method.